lasagne/tests/layers/test_pool.py

```diff
@@ -60,6 +60,13 @@ def upscale_1d(data, scale_factor):
     return upscaled
 
 
+def upscale_1d_bed_of_nails(data, scale_factor):
+    upscaled = np.zeros(upscale_1d_shape(data.shape, scale_factor))
+    _, _, i = np.nonzero(data)
+    upscaled[:, :, i * scale_factor[0]] = data[:, :, i]
```

Comment:
Again, the `nonzero` makes it more expensive. Just use ordinary slicing: `upscaled[..., ::scale_factor[0]] = data`